source/modulo_controllers/README.md

```diff
@@ -28,3 +28,64 @@ initial states to prevent any NaN data from propagating through to the control l
+joint command or access the joint state using the `set_joint_command()` and `get_joint_state()` methods, respectively.
+
+### Robot model and URDF
+
+If a controller needs a robot model, a `robot_model::Model` will be configured from URDF information. To support this,
+the `robot_description` must be specified.
+
+The robot model is used to calculate the CartesianState of a task frame from the JointState using forward kinematics,
+which is available to derived classes using the `get_cartesian_state()` method.
+
+The task frame can be specified using the `task_space_frame` parameter. If left empty, the final link of the
+robot model will be used.
+
+In certain cases, the `joints` parameter might not be in a physically correct order; for example, they might be sorted
+alphabetically, which makes the corresponding JointState incompatible with the actual order of joints in a robot system.
+If set to true, the `sort_joints` parameter orders the joint names to be physically correct using the robot model data.
+
+Derived controllers can also access and leverage the robot model for more advanced behaviors such as inverse kinematics.
+
+### Force-Torque Sensor
+
+If the `ft_sensor_name` and `ft_sensor_reference_frame` parameters are set, the controller will look for a matching
+sensor with the following state interfaces:
+
+- `force.x`
+- `force.y`
+- `force.z`
+- `torque.x`
+- `torque.y`
+- `torque.z`
+
+Derived controllers can use `get_ft_sensor()` to access a CartesianWrench object of the sensor data. If the
+`ft_sensor_reference_frame` matches the `task_space_frame`, the measured wrench will additionally be transformed to
+the robot base frame and applied to the Cartesian state of the robot that is available through `get_cartesian_state()`.
+
+### General behavior
+
+The `command_half_life` parameter is used to attenuate any dynamic commands (velocities, accelerations or torques) in
+case no new commands are received (i.e. if `set_joint_command()` was not called since the last controller update).
+
+At each control step, the last command is scaled by a decay factor to cause an exponential fall-off towards zero. The
+decay factor is calculated in terms of the desired command half-life based on the control frequency.
+At the half-life time, the command value will be reduced by half. For example, a controller with a command half-life
+of 0.1 seconds will have the output command reduced by 50%, 25%, 12.5% and 6.25% at 0.1, 0.2, 0.3 and 0.4 seconds
+respectively.
+
+The `command_rate_limit` parameter sets an upper bound on how much a command interface value can change at each
+control step. It is expressed in units per second. For example, a controller with a rate limit of 5.0 commanding the
+joint velocity interface at 100Hz (a control period of 0.01 seconds) will only allow the joint velocity to change by
+up to 0.05 radians per second at each control step, or at 5.0 radians per second, per second.
```

Comment:
```suggestion
up to 0.05 radians per second at each control step, or at 5.0 radians per second squared.
```

right?

Reply:
I think the Doc found it better this way - or even wrote it himself because that's not something I'd write :smile:

Reply:
Then this is beyond me and my English :stuck_out_tongue:, and  in any case it's correct as is